src/components/navigation/z-menu-section/index.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  Component,
+  h,
+  Prop,
+  State,
+  Event,
+  EventEmitter,
+  Element,
+  Listen,
+  Host
+} from '@stencil/core';
+
+/**
+ * @slot - Menu section label
+ * @slot item - Single entry of the section. Can be slotted multiple times to insert items.
+ */
+
+@Component({
+  tag: 'z-menu-section',
+  styleUrl: 'styles.css',
+  shadow: true
+})
+
+export class ZMenuSection {
+  @Prop({ reflect: true }) active?: boolean;
+  @State() open: boolean;
+  @State() hasContent: boolean;
+  @Element() hostElement: HTMLElement;
+
+  /** The section has been opened. */
+  @Event() opened: EventEmitter;
+  /** The section has been closed. */
+  @Event() closed: EventEmitter;
+  toggle() {
+    if (!this.hasContent) {
+      return;
+    }
+
+    this.open = !this.open;
+    this.open ? this.opened.emit() : this.closed.emit();
+  }
+
+  @Listen('click', { target: 'document' })
+  /** Close the list when a click is performed outside of this Element. */
+  handleClick(ev) {
+    if (!this.open || this.hostElement.contains(ev.target)) {
+      return;
+    }
+
+    this.toggle();
```

Comment:
I would prefer to set the open state to false instead of using the toggle method.